scipy/special/specfun/specfun.f

```diff
@@ -6228,6 +6230,14 @@ SUBROUTINE HYGFZ(A,B,C,Z,ZHF,ISFER)
            CALL GAMMA2(1.0D0+A/2.0D0-B,G2)
            CALL GAMMA2(0.5D0+0.5D0*A,G3)
            ZHF=G0*G1/(G2*G3)
+        ELSE IF (L7.AND.((A.EQ.C).OR.(B.EQ.C))) THEN
+           IF (A.EQ.C) A=BB
```

Comment:
That modifies the input argument --- probably should set `AA=BB` and use `AA` instead below.

Also, is the intent of if statements that a=c=negint or b=c=negint?
If yes, that does not seem to be guaranteed by the above.